doc/reference/tx-metadata.md

```diff
@@ -1,16 +1,75 @@
+
+There is no limit on the number of key-value pairs (except the overall transaction size limit) or on individual structured values. However, there *is* a limit on the size of text strings and byte strings within the structured values, which is implemented to mitigate the problem of unpleasant or illegal content being posted to the blockchain.
+
+Metadata *does not require* any additional fees. It simply contributes to the size of the transaction with a one-time processing fee based on transaction size.
+
+**Binary schema**
+
+The binary schema is based on the Concise Binary Object Representation ([CBOR)](https://tools.ietf.org/html/rfc7049) and Concise Data Definition Language ([CDDL)](https://tools.ietf.org/html/rfc8610) notations, and is presented as follows:
+
+```
+transaction_metadatum =
+    { * transaction_metadatum => transaction_metadatum }
+  / [ * transaction_metadatum ]
+  / int
+  / bytes .size (0..64)
+  / text .size (0..64)
+
+transaction_metadatum_label = uint
+
+transaction_metadata =
+  { * transaction_metadatum_label => transaction_metadatum }
+```
+
+## How to create a transaction with metadata using the cardano-cli
+
+To create a transaction with metadata, first ensure that you have installed the [cardano-node](https://github.com/input-output-hk/cardano-node#cardano-node-overview) and follow instructions on how to launch [cardano-CLI](https://github.com/input-output-hk/cardano-node/tree/master/cardano-cli#cardano-cli).
+
+To create a transaction with specified metadata, run this command:
+
+`cabal exec cardano-cli -- shelley transaction build-raw`
+
+Shelley transactions can be created with the following options:
+
++ according to a chosen schema: *no schema* or *detailed schema*
++ according to a chosen metadata format: CBOR or JSON
+
+```
+  --json-metadata-no-schema
+                       	Use the "no schema" conversion from JSON to tx
+                       	metadata.
+  --json-metadata-detailed-schema
+                       	Use the "detailed schema" conversion from JSON to tx
+                       	metadata.
+  --metadata-json-file FILE
+                       	Filepath of the metadata file, in JSON format.
+  --metadata-cbor-file FILE
+                       	Filepath of the metadata, in raw CBOR format.
+```
+
+## Metadata schemas - mappings and formats
+
+On-chain metadata that is carried along with transactions is encoded according to CBOR. To create a transaction, you can add the metadata with pre-encoded CBOR. Alternatively, you can add metadata in JSON format, which will be converted into the internal format. In this section, we provide examples of the two different mappings: tx metadata and JSON, which are useful for different purposes. In both schemas, the top-level JSON is an object indexed by integers, which are mapped to JSON values.
```

Comment:
Instead of 
> In this section, we provide examples of the two different mappings: tx metadata and JSON

I would say:
> In this section, we provide examples of the two different mappings _between_ tx metadata and JSON